internal/profile/install_profile.go

```diff
@@ -0,0 +1,194 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package profile
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/elastic-package/internal/configuration/locations"
+	"github.com/elastic/elastic-package/internal/logger"
+)
+
+// CreateProfile creates an existing profile from the default elastic-package config dir
+// if option.Package has a value it'll be used, if not, the default location will be used
+// if option.From does not have a supplied value, it'll create a default profile.
+func CreateProfile(options Options) error {
+	if options.PackagePath == "" {
+		loc, err := locations.NewLocationManager()
+		if err != nil {
+			return errors.Wrap(err, "error finding stack dir location")
+		}
+		options.PackagePath = loc.StackDir()
+	}
+
+	if options.FromProfile == "" {
+		return createProfile(options)
+	}
+
+	return createProfileFrom(options)
+}
+
+// createProfile installs a new profile at the given package path location.
+// overwriteExisting determines the behavior if a profile with the given name already exists.
+// On true, it'll overwrite the profile, on false, it'll copy the existing profile over to profileName_old
+func createProfile(options Options) error {
+	profile, err := NewConfigProfile(options.PackagePath, options.Name)
+	if err != nil {
+		return errors.Wrap(err, "error creating profile")
+	}
+
+	//check to see if we have an existing profile at that location.
+	exists, err := profile.alreadyExists()
+	if err != nil {
+		return errors.Wrap(err, "error checking for existing profile")
+	}
+	if exists {
+		localChanges, err := profile.localFilesChanged()
+		if err != nil {
+			return errors.Wrapf(err, "error checking for changes in %s", profile.ProfilePath)
+		}
+
+		// If there's changes and we've selected CreateNew, move the old path
+		// TODO: do we want this to pe appended with some kind of version string instead?
+		if localChanges && !options.OverwriteExisting {
+			if localChanges && options.Name == DefaultProfile {
+				logger.Warn("WARNING: default profile has been changed by user or updated by elastic-package. The current profile will be moved to default_old.")
```

Comment:
IMO, we can drop the `WARNING` prefix as the `logger.Warn` method will already print `WARN`.
```suggestion
				logger.Warn("default profile has been changed by user or updated by elastic-package. The current profile will be moved to default_old.")
```